llvm/include/llvm/IR/FPAccuracy.def

```diff
@@ -0,0 +1,63 @@
+//===--- llvm/IR/FPAccuracy.def - Mappings for fp accuracy  -----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Defines properties of floating point builtin intrinsics.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FP_ACCURACY
+#define FP_ACCURACY(IID,Sf",SD,Cf",CD)
+#endif
+
+// Each entry below maps an fpbuiltin intrinsic ID to the required accuracy
+// for that operation for single- and double-precision for SYCL and CUDA
+//
+// All accuracies are returned as single-precision floating-point values.
+//
+// Note: for single-precision fdiv and sqrt, the value returned here assumes
+// that the  "-cl-fp32-correctly-rounded-divide-sqrt " option is not used.
```

Comment:
I've updated the comment. I'm not aware of any current plans to define accuracy levels for other models. I'm not sure what requirements other platforms have. In fact, I think even the CUDA accuracy is a statement of the actual accuracy provided by CUDA's math library and not a "requirement" as such. I think its usefulness here is for people who aren't using CUDA but want results that are as accurate as they would have gotten with CUDA. The SYCL level defined here is actually derived from the OpenCL standard, and it would also apply to OpenMP implementations that use OpenCL as their offload mechanism. I couldn't find any accuracy requirement documentation for HIP.